python/pyarrow/table.pxi

```diff
@@ -671,13 +671,55 @@ cdef class RecordBatch(_PandasConvertible):
         Returns
         -------
         RecordBatch
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> pydict = {'int': [1, 2], 'str': ['a', 'b']}
+        >>> pa.RecordBatch.from_pydict(pydict)
+        pyarrow.RecordBatch
+        int: int64
+        str: string
         """
 
         return _from_pydict(cls=RecordBatch,
                             mapping=mapping,
                             schema=schema,
                             metadata=metadata)
 
+    @staticmethod
+    def from_pylist(mapping, schema=None, metadata=None):
+        """
+        Construct a RecordBatch from Arrow arrays or columns.
+
+        Parameters
+        ----------
+        mapping : list of dicts or Mappings
+            A mapping of strings to Arrays or Python lists.
+        schema : Schema, default None
+            If not passed, will be inferred from the Mapping values.
+        metadata : dict or Mapping, default None
+            Optional metadata for the schema (if inferred).
+
+        Returns
+        -------
+        RecordBatch
+
+        Examples
+        --------
+        >>> import pyarrow as pa
+        >>> pylist = [{'int': [1, 2]}, {'str': ['a', 'b']}]
+        >>> pa.RecordBatch.from_pylist(pylist)
```

Comment:
Actually, I see the difference now (this is a list of separate dict per column instead of a single dict). But, I think this is actually not exactly what we want here. Or at least, my expectation was to be able to handle an input like:

```
pylist = [{'int': 1, 'str'; 'a'}, {'int': 2, 'str': 'b'}]
```

So also a list of dict, but differently organized (one dict per row, instead of one dict per column)

Reply:
Agree, As a user I would personally find more obvious that it accepts a list of rows whhen building a RecordBatch or Table.

Reply:
Thanks for the additional explanation! Will correct (in the 2022 I think =)